### PR TITLE
RestAssured is ready

### PR DESCRIPTION
**Pull request by AleksandrDark**
Please, take a look at my work

### CODE REVIEW

File: src/main/java/dto/CardTrelloDTO.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package dto;
+
+import lombok.Data;
+
+@Data
+public class CardTrelloDTO {
+    private String id;
+    private String name;
+    private String idList;
+
```
Comment (GrebNat) [inline src/main/java/dto/CardTrelloDTO.java]: empty line

File: src/main/java/dto/ListTrelloDTO.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package dto;
+
+import lombok.Data;
+
+@Data
+public class ListTrelloDTO {
+
```
Comment (GrebNat) [inline src/main/java/dto/ListTrelloDTO.java]: empty line

File: src/main/java/service/BoardTrelloService.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package service;
+
+import com.google.gson.GsonBuilder;
+import dto.BoardTrelloDTO;
+import io.restassured.http.Method;
+import io.restassured.response.Response;
+import java.util.Map;
+import util.URI;
+
+
```
Comment (GrebNat) [inline src/main/java/service/BoardTrelloService.java]: empty line

File: src/main/java/service/BoardTrelloService.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+public class BoardTrelloService extends CommonService {
+
+    private BoardTrelloDTO parseBoardTrello(Response response) {
```
Comment (GrebNat) [inline src/main/java/service/BoardTrelloService.java]: move private to the bottom

File: src/main/java/service/BoardTrelloService.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    public BoardTrelloDTO updateBoardTrello(Map<String, String> params, String idBoard) {
+        return parseBoardTrello(requestWithParams(Method.PUT, URI.ENDPOINT_BOARD + idBoard, params));
+    }
+
+    public Response deleteBoardTrello(String idBoard) {
+        return requestNoParams(Method.DELETE, URI.ENDPOINT_BOARD + idBoard);
+    }
+
```
Comment (GrebNat) [inline src/main/java/service/BoardTrelloService.java]: empty line

File: src/main/java/service/CardTrelloService.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import com.google.gson.GsonBuilder;
+import dto.CardTrelloDTO;
+import io.restassured.http.Method;
+import io.restassured.response.Response;
+import java.util.Map;
+import util.URI;
+
+public class CardTrelloService extends CommonService {
+
+    private CardTrelloDTO parseCardTrello(Response response) {
```
Comment (GrebNat) [inline src/main/java/service/CardTrelloService.java]: there is same method for board. could be moved to CommonService

File: src/main/java/service/CommonService.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+    private final RequestSpecification REQUEST_SPECIFICATION;
+
+    public CommonService() {
+        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+        REQUEST_SPECIFICATION = new RequestSpecBuilder()
+            .setBaseUri(URI.BASE_URL)
+            .setContentType(ContentType.JSON)
+            .setAccept(ContentType.JSON)
+            .addQueryParam("key", PropertyReader.getProperties().get("trelloKey").toString())
```
Comment (GrebNat) [inline src/main/java/service/CommonService.java]: update to 
PropertyReader.getTrelloKey()

File: src/main/java/service/CommonService.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+            given(REQUEST_SPECIFICATION)
+                .queryParams(params)
+            .when()
+                .request(method, uri)
+            .then()
+                .statusCode(SC_OK)
+                .extract()
+                .response();
+    }
+
```
Comment (GrebNat) [inline src/main/java/service/CommonService.java:50]: empty line

File: src/main/java/service/CommonService.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        return
+            given(REQUEST_SPECIFICATION)
+                .queryParams(params)
+            .when()
+                .request(method, uri)
+            .then()
+                .statusCode(SC_OK)
+                .extract()
+                .response();
+    }
```
Comment (GrebNat) [inline src/main/java/service/CommonService.java:49]: code duplication for in 2 methods above. call one method from other with empty params.

File: src/main/java/util/URI.java [hunk 10]
```diff
@@ -0,0 +1,10 @@
+package util;
+
+public class URI {
+
+    public static final String BASE_URL = "https://api.trello.com/";
```
Comment (GrebNat) [inline src/main/java/util/URI.java]: move to tests.properties

File: src/main/java/service/ListTrelloService.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+    public ListTrelloDTO getListTrello(String idList) {
+        return parseBoardTrello(requestNoParams(Method.GET, URI.ENDPOINT_LIST + idList));
+    }
+
+    public ListTrelloDTO updateListTrello(Map<String, String> params, String idList) {
+        return parseBoardTrello(requestWithParams(Method.PUT, URI.ENDPOINT_LIST + idList, params));
+    }
+
+}
```
Comment (GrebNat) [inline src/main/java/service/ListTrelloService.java:23]: same comments like for Card... class

File: src/main/java/service/CommonService.java (hunk 9 above)
Comment (GrebNat) [inline src/main/java/service/CommonService.java:49]: code duplication for in 2 methods above. call one method from other with empty params.

File: src/main/java/util/URI.java [hunk 12]
```diff
@@ -0,0 +1,10 @@
...
+    public static String ENDPOINT_BOARD = "1/boards/";
```
Comment (GrebNat) [inline src/main/java/util/URI.java:5]: separate board/card/list endpoint for  CREATE and GET 
 "1/boards/"
  "1/boards/{0}"

File: src/test/java/assertions/AssertionsTrello.java [hunk 13]
```diff
@@ -0,0 +1,30 @@
+package assertions;
+
+import dto.BoardTrelloDTO;
+import dto.CardTrelloDTO;
+import dto.ListTrelloDTO;
+import io.restassured.response.Response;
+import org.assertj.core.api.Assertions;
+
+public class AssertionsTrello {
+
```
Comment (GrebNat) [inline src/test/java/assertions/AssertionsTrello.java]: add error messages for all assetiongs

Comment (GrebNat) [inline src/test/java/assertions/AssertionsTrello.java]: move assertion class into the main

File: src/test/java/functionalteststrello/TrelloTest.java [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+
+import io.restassured.response.Response;
+import org.testng.annotations.Test;
+import data.DataTest;
+
+public class TrelloTest extends BaseTest {
+
+    @Test
+    public void checkBoardCreatedAndGettingAndDeleted() {
+        String idBoard = boardTrelloDTO.getId();
```
Comment (GrebNat) [inline src/test/java/functionalteststrello/TrelloTest.java]: test naming and test steps are inconsistent

File: src/test/java/functionalteststrello/TrelloTest.java [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+        boardTrelloDTO = boardTrelloService.getBoardTrello(idBoard);
+        assertionsTrello.checkBoardName(boardTrelloDTO, DataTest.BOARD_NAME);
+    }
+
+    @Test
+    public void checkCreatedAndUpdatedBoard() {
+        String idBoard = boardTrelloDTO.getId();
```
Comment (GrebNat) [inline src/test/java/functionalteststrello/TrelloTest.java:20]: board should be created into the test. There should not be any hidden steps

**Review by GrebNat — CHANGES_REQUESTED** [review]

File: src/main/java/util/PropertyReader.java [hunk 16]
```diff
@@ -0,0 +1,27 @@
+package util;
+
+import java.util.Properties;
+import lombok.SneakyThrows;
+
+public class PropertyReader {
+
+    public static String getKey() {
+        return PropertyReader.getProperties("properties/conf.properties").get("trelloKey").toString();
```
Comment (GrebNat) [inline src/main/java/util/PropertyReader.java]: pls, do not read file with properties every time. Read it once, at the first request

File: src/test/resources/properties/test.properties [hunk 17]
```diff
@@ -0,0 +1 @@
+BASE_URL=https://api.trello.com/
```
Comment (GrebNat) [inline src/test/resources/properties/test.properties]: it is ok to combine conf.properties and test.properties files

File: src/test/java/functionalteststrello/TrelloTest.java [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+        assertionsTrello.checkBoardName(boardTrelloDTO, DataTest.BOARD_NAME);
+    }
+
+    @Test
+    public void checkUpdatedBoard() {
+        boardParams.put("name", DataTest.BOARD_NAME);
+        boardTrelloDTO = boardTrelloService.createBoardTrello(boardParams);
+        String idBoard = boardTrelloDTO.getId();
+        boardTrelloService.getBoardTrello(idBoard);
+        System.out.println(boardTrelloService);
```
Comment (GrebNat) [inline src/test/java/functionalteststrello/TrelloTest.java]: why ??

**Review by GrebNat — CHANGES_REQUESTED** [review]

File: src/test/java/functionalteststrello/TrelloTest.java (hunk 18 above)
Comment (AleksandrDark) [reply src/test/java/functionalteststrello/TrelloTest.java]: it is my mistake. I forgot to delete the unnecessary lines

**Review by GrebNat — APPROVED** [review]

**Pull request merged**